agents/holo_synth/synth_agent.py

```diff
@@ -0,0 +1,234 @@
+            if not acquired:
+                self.log.warn(
+                    f"Could not set position because lock held by {self.lock.job}"
+                )
+                return False, "Could not acquire lock"
+
+        return True, "Frequencies Updated"
+
+    @ocs_agent.param("lo_id_n", type=int, default=0, check=lambda x: (x == 0) or (x == 1))
+    @ocs_agent.param("status", type=int, default=0, check=lambda x: (x == 0) or (x == 1))
+    def set_synth_status(self, session, params):
+        """set_synth_status(lo_id_n=0, status=1)
+
+        **Task** - A task to set the status of the synthesizers.
+
+        Parameters:
+            dict: {'lo_id_n': int
+                   'status': int}
+        Examples:
+            Example for calling in a client::
+
+                import ocs
+                from ocs.ocs_client import OCSClient
+                agent = OCSClient("synth_LO")
+                agent.set_synth_status(lo_id_n=0, status=1)
+        Notes:
+            Set the status of the synthesizers either on (1) or off (0).
+        """
+        lo_id_n = params.get("lo_id", 0)
+        switch = params.get("switch", 0)
+
+        with self.lock.acquire_timeout(timeout=3, job="turn_on_or_off_synth") as acquired:
+            if not acquired:
+                self.log.warn(
+                    f"Could not set position because lock held by {self.lock.job}"
+                )
+                return False, "Could not acquire lock"
+
+            synth3.set_RF_output(lo_id_n, switch, self.lo_id)
+
+            data = {"timestamp": time.time(), "block_name": "synth_LO", "data": {}}
+
+            data["data"]["F1_status"] = switch
+
+            self.agent.publish_to_feed("synth_LO", data)
+            session.data.update(data["data"])
+
+        return True, "Frequencies Updated"
+
+
+def make_parser(parser=None):
+    """Build the argument parser for the Agent. Allows sphinx to automatically
+    build documentation based on this function.
+    """
+    if parser is None:
+        parser = argparse.ArgumentParser()
+
+    # Add options specific to this agent.
+    pgroup = parser.add_argument_group("Agent Options")
+    pgroup.add_argument("--config_file")
+
+    return parser
+
+
+if __name__ == "__main__":
+    # For logging
+    txaio.use_twisted()
+    LOG = txaio.make_logger()
+
+    # Start logging
+    txaio.start_logging(level=os.environ.get("LOGLEVEL", "info"))
+
+    parser = make_parser()
+
+    # Interpret options in the context of site_config.
+    args = site_config.parse_args(agent_class="SynthAgent", parser=parser)
+
+    agent, runner = ocs_agent.init_site_agent(args)
+
+    synth_agent = SynthAgent(agent, args.config_file)
+
+    agent.register_task("init_synth", synth_agent.init_synth)
+    agent.register_task("set_frequencies", synth_agent.set_frequencies)
+    agent.register_task("read_frequencies", synth_agent.read_frequencies)
```

Comment:
Since `read_frequences` isn't functional, let's not register it as a task yet. Comment this line out.